app/controllers/opinions_controller.rb

```diff
@@ -0,0 +1,27 @@
+class OpinionsController < ApplicationController
+
+  def create
+    @article = Article.published.find(params[:article_id])
+    @comment = Comment.find(params[:comment_id])
+    @opinion = Opinion.new(opinion_params)
+    @opinion.user = current_user if current_user
+    @opinion.comment = @comment
+    @opinion.comment.article = @article
+
+    @opinion.save
+
+    redirect_to article_path(@article)
+  end
+
+  def destroy
+    @opinion = Opinion.find(params[:id])
+    @opinion.destroy
+    redirect_to article_comments_path(opinion.article)
+  end
+
+  private
+  def opinion_params
+     params.require(:opinion).permit(:opinion, :user)
```

Comment:
Czy user jest tu potrzebny?

Reply:
Nie mogłam zapisać opinii do bazy bez usera, dlatego że mam połączoną opinię z userem (w bazie foreign key, a w aplikacji belongs to).